File: src/main/java/com/paragon/client/systems/ui/taskbar/Taskbar.java
package com.paragon.client.systems.ui.taskbar;

import com.paragon.api.util.Wrapper;
import com.paragon.api.util.render.GuiUtil;
import com.paragon.api.util.render.RenderUtil;
import com.paragon.api.util.render.TextRenderer;
import com.paragon.client.systems.module.impl.client.ClientFont;
import com.paragon.client.systems.ui.console.ConsoleGUI;
import com.paragon.client.systems.ui.taskbar.icons.Icon;
import com.paragon.client.systems.module.impl.client.Colours;
import com.paragon.client.systems.module.impl.client.ClickGUI;
import net.minecraft.client.gui.ScaledResolution;

import java.awt.*;
import java.util.ArrayList;

/**
 * @author Wolfsurge
 */
public class Taskbar implements Wrapper, TextRenderer {

    private ArrayList<Icon> icons = new ArrayList<>();
    private boolean open = true;

    public Taskbar() {
        int x = (int) getStringWidth("Paragon") + 10;

        icons.add(new Icon("GUI", x, ClickGUI::getGUI));
        x += getStringWidth("GUI") + 7;
        icons.add(new Icon("Console", x, ConsoleGUI::new));
    }

    public void drawTaskbar(int mouseX, int mouseY) {
        ScaledResolution scaledResolution = new ScaledResolution(mc);

        if (open) {
            RenderUtil.drawRect(0, scaledResolution.getScaledHeight() - 21, scaledResolution.getScaledWidth(), 21, new Color(20, 20, 20).getRGB());
            RenderUtil.drawRect(0, scaledResolution.getScaledHeight() - 23, scaledResolution.getScaledWidth(), 2, Colours.mainColour.getColour().getRGB());

            renderText("Paragon", 2, scaledResolution.getScaledHeight() - 15, Colours.mainColour.getColour().getRGB());

            for (Icon icon : icons) {
                icon.draw(mouseX, mouseY);
            }
        }

        renderText(open ? "Close" : "Open", scaledResolution.getScaledWidth() - getStringWidth(open ? "Close" : "Open") - 3, scaledResolution.getScaledHeight() - getFontHeight() - (ClientFont.INSTANCE.isEnabled() ? 3 : 5), -1);
    }

    public void mouseClicked(int mouseX, int mouseY) {
        ScaledResolution scaledResolution = new ScaledResolution(mc);

        if (open) {
            icons.forEach(icon -> {
                icon.whenClicked(mouseX, mouseY);
            });
        }

        if (GuiUtil.mouseOver(scaledResolution.getScaledWidth() - getStringWidth(open ? "Close" : "Open") - 3, scaledResolution.getScaledHeight() - getFontHeight() - 5, scaledResolution.getScaledWidth(), scaledResolution.getScaledHeight(), mouseX, mouseY)) {
            open = !open;
        }
    }

}

File: src/main/kotlin/com/paragon/impl/ui/taskbar/Taskbar.kt
package com.paragon.impl.ui.taskbar

import com.paragon.util.render.font.FontUtil
import com.paragon.impl.module.client.Colours
import com.paragon.impl.ui.taskbar.start.StartMenu
import com.paragon.impl.ui.util.Click
import com.paragon.util.Wrapper
import com.paragon.util.render.RenderUtil
import me.surge.animation.Animation
import me.surge.animation.Easing
import net.minecraft.client.gui.ScaledResolution
import java.awt.Color

/**
 * @author Surge
 * @since 26/07/2022
 */
object Taskbar : Wrapper {

    var tooltip = ""

    val expandAnimation: Animation = Animation({ 500f }, false, { Easing.CUBIC_IN_OUT })
    val timer: Animation = Animation({ 1000f }, false, { Easing.LINEAR })

    fun draw(mouseX: Int, mouseY: Int) {
        val scaledResolution = ScaledResolution(minecraft)
        val hovered = mouseY.toFloat() in scaledResolution.scaledHeight - 80f..scaledResolution.scaledHeight.toFloat()

        var state = false

        if (StartMenu.expandAnimation.state || hovered) {
            state = true
        }

        expandAnimation.state = state

        RenderUtil.drawRect(
            0f, scaledResolution.scaledHeight - (26f * expandAnimation.getAnimationFactor().toFloat()), scaledResolution.scaledWidth.toFloat(), 26f, Color(148, 148, 148).rgb
        )

        StartMenu.x = 2f
        StartMenu.y = scaledResolution.scaledHeight - (23f * expandAnimation.getAnimationFactor().toFloat())
        StartMenu.draw(mouseX, mouseY)

        RenderUtil.drawRect(
            150f, scaledResolution.scaledHeight - (26f * expandAnimation.getAnimationFactor().toFloat()) - 2, scaledResolution.scaledWidth.toFloat() - 150f, 2f, Colours.mainColour.value.rgb
        )

        if (tooltip != "") {
            RenderUtil.drawRect(
                (scaledResolution.scaledWidth - FontUtil.getStringWidth(tooltip) - 4) + ((FontUtil.getStringWidth(
                    tooltip
                ) + 4) * expandAnimation.getAnimationFactor()).toFloat(), scaledResolution.scaledHeight - 16f, FontUtil.getStringWidth(tooltip) + 4, 13f, 0x90000000.toInt()
            )
            FontUtil.drawStringWithShadow(
                tooltip, scaledResolution.scaledWidth - FontUtil.getStringWidth(tooltip) - 2, scaledResolution.scaledHeight - 14f, Color.WHITE.rgb
            )
        }
    }

    fun mouseClicked(mouseX: Int, mouseY: Int, click: Click) {
        if (!StartMenu.mouseClicked(mouseX, mouseY, click)) {
            StartMenu.expandAnimation.state = false
        }
    }

}